### DISCUSSION TITLE
Global Chai Declarations

### DISCUSSION THREAD

**Discussion by kpowerz** [cypress-io/cypress#9618] [Questions and Help] [answered]
Please excuse me for I've only been using cypress for a month now, but I was curious as to why cypress declares the expect and assert globally? 

Isn't there a way where the expect and assert can be declared on a 'cypress' level'? As in only within the cypress folder?

Obviously, this is something I can resolve myself but I was curious as to why this is done by default.

**Comment by bahmutov** [comment, marked as answer]
Well, it would be nice to limit the number of global variables, agree. But
historically every spec would need to
import `cypress` and `expect` and whatever else. At the start we have
decided to make it easier to do testing
by having everything ready to go.

Maybe with time we could transition to being local and explicit like

const {it, describe, expect, cy} = require('cypress')
describe('my tests', () => {
  it('...', () => {
     expect(1).to.equal(1)
  })
})

On Thu, Dec 10, 2020 at 11:32 AM Kyle <notifications@github.com> wrote:

> Please excuse me for I've only been using cypress for a month now, but I
> was curious as to why cypress declares the expect and assert globally?
>
> Isn't there a way where the expect and assert can be declared on a
> 'cypress' level'? As in only within the cypress folder?
>
> Obviously, this is something I can resolve myself but I was curious as to
> why this is done by default.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cypress-io/cypress/discussions/9618>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAQ4BJRXVA3QFI74UVYVNITSUDZ2ZANCNFSM4UVIN3QQ>
> .
>


-- 
Dr. Gleb Bahmutov, PhD

Schedule video chat / phone call / meeting with me via
https://calendly.com/bahmutov
gleb.bahmutov@gmail.com @bahmutov <https://twitter.com/@bahmutov>
https://glebbahmutov.com/ https://glebbahmutov.com/blog
https://github.com/bahmutov

**Reply by kpowerz** [reply]
Awesome, that is great to hear. I've been using cypress full time at my new job and have been really enjoying it. Excited to see its continuous evolution!

Thanks for the reply.
